src/main/java/org/broadinstitute/hellbender/ZipUtils.java

```diff
@@ -0,0 +1,128 @@
+package org.broadinstitute.hellbender.utils;
+
+import org.apache.commons.io.IOUtils;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+
+import java.io.*;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.zip.*;
+
+/**
+ * Utility class to zip and unzip files.
+ */
+public class ZipUtils {
```

Comment:
Zip is just a convenient format to have more than one file within the file. An alternative would be to have a single file with a very rich header. Why bother the user with unzipping or zipping he does not need to care about whether is a zip or a simple text file. Did you know that Word .docx are actually zips?

Reply:
If we transform this into a feature file (now or eventually) then is going to be the rich header approach.

Reply:
We have examples of side inputs for feature files for other formats. VCF.gz files can have several types of indexes for example and i think something in line with those indexes could be accomplished. I just think it might be useful in the future to open these up as features since then we can use existing machinery for indexing/caching etc and we could conceivably make something like a strSiteWalker etc for other use cases.

Reply:
Will create a ticket for this one.